R/FilterStatesMAE.R

```diff
@@ -259,21 +257,15 @@ MAEFilterStates <- R6::R6Class( # nolint
         id = id,
         function(input, output, session) {
           logger::log_trace("MAEFilterState$srv_add initializing, dataname: { private$dataname }")
-          active_filter_vars <- reactive({
-            vapply(
-              X = private$state_list_get("y"),
-              FUN.VALUE = character(1),
-              FUN = function(x) x$get_varname()
-            )
-          })
 
           # available choices to display
           avail_column_choices <- reactive({
-            choices <- setdiff(get_supported_filter_varnames(data = data), active_filter_vars())
+            active_filter_vars <- unique(unlist(extract_feat(self$get_filter_state(), "varname")))
+            choices <- setdiff(get_supported_filter_varnames(data = data), active_filter_vars)
             varlabels <- vapply(
               colnames(data),
               FUN = function(x) {
-                label <- attr(data[[x]], "label")
+                label <- attr(data[[x]], "label") # TODO: can this ever be NULL or just character(0)
```

Comment:
If attribute isn't specified then it returns NULL. If you want this to be robust then line below we should have `if (!checkmate::test_string(label))`.

```suggestion
                label <- attr(data[[x]], "label")
```